tests/LSTM_baseline_classifier.py

```diff
@@ -0,0 +1,116 @@
+from torch.utils.data import Dataset, DataLoader
+import sys
+from sklearn.metrics import f1_score
+import torch
+import matplotlib.pyplot as plt
+import torch.nn as nn
+
+sys.path.append('/path_to_directory_containing_HAHA/')
+
+from oai_agents.common.overcooked_dataset_Baseline.py import main as initialize_overcooked_dataset  # Import the main function
+
+OD = initialize_overcooked_dataset()
+distinct_trial_ids = list(set(OD.main_trials['trial_id']))
+distinct_trial_ids.sort()  # Ensure they are in order
+
+# Split trial IDs at the 85% mark
+split_index = int(0.85 * len(distinct_trial_ids))
+train_trial_ids = distinct_trial_ids[:split_index]
+test_trial_ids = distinct_trial_ids[split_index:]
+
+# Split the dataset based on these trial IDs
+train_indices = [index for index, trial_id in enumerate(OD.main_trials['trial_id']) if trial_id in train_trial_ids]
+test_indices = [index for index, trial_id in enumerate(OD.main_trials['trial_id']) if trial_id in test_trial_ids]
+
+train_data = [OD[idx] for idx in train_indices]
+test_data = [OD[idx] for idx in test_indices]
+
+# Create DataLoaders
+train_loader = DataLoader(train_data, batch_size=64, shuffle=True, drop_last=True)
+val_loader = DataLoader(test_data, batch_size=64, shuffle=False, drop_last=True)
+
+class LSTMClassifier(nn.Module):
+    def __init__(self, input_size, hidden_size, output_size, num_layers):
+        super(LSTMClassifier, self).__init__()
+        self.lstm = nn.LSTM(input_size, hidden_size, num_layers, batch_first=True)
+        self.fc = nn.Linear(hidden_size, output_size)
+
+    def forward(self, x):
+        out, _ = self.lstm(x)
+        out = out[:, -1, :]
```

Comment:
You are only predicting using the last output, but you can actually predict at using the output at every timestep to improve the learning signal (during testing, you would only use the last one since that'll be the best prediction).